airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -83,6 +83,15 @@ def create_session(self) -> boto3.session.Session:
                 self.extra_config["session_kwargs"],
             )
             session_kwargs = self.extra_config["session_kwargs"]
+
+        if "profile" in self.extra_config and "s3_config_file" not in self.extra_config:
```

Comment:
Why initially it added in AWS Hook or why I add this warning?

Answer for fist part: https://github.com/apache/airflow/pull/2532#discussion_r145792572 
Answer for second part: This parameter exist in documentation since version 1.1.0 without mentioning how it actually works, I've just add warning to inform user.